File: CS/Q205964/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGauges.Win;
using DevExpress.XtraGauges.Win.Gauges.Circular;
using DevExpress.XtraGauges.Base;
using DevExpress.XtraGauges.Core.Model;
using DevExpress.XtraGauges.Core.Drawing;

namespace Q205964 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            CreateCircularGauge();
        }

        private void CreateCircularGauge() {
            GaugeControl gaugeControl = new GaugeControl();
            CircularGauge gauge = (CircularGauge)gaugeControl.AddGauge(GaugeType.Circular);
            gauge.AddDefaultElements();
            ArcScaleComponent scale = gauge.Scales[0];
            scale.BeginUpdate();
            scale.Ranges.AddRange(CreateRanges(scale.MaxValue - scale.MinValue, scale.MinValue));
            scale.EndUpdate();
            gaugeControl.Dock = DockStyle.Fill;
            Controls.Add(gaugeControl);
        }

        private Color[] rangeContentColors = { Color.Red, Color.Yellow, Color.Green };

        private IRange[] CreateRanges(float ticks, float start) {
            int range = (int)Math.Floor((double)(ticks / rangeContentColors.Length));
            List<IRange> ranges = new List<IRange>();
            for (int i = 0; i < rangeContentColors.Length; )
                ranges.Add(CreateRange(rangeContentColors[i], (int)(range * i + start), (int)(range * ++i + start)));
            return ranges.ToArray();
        }

        private IRange CreateRange(Color contentColor, int startValue, int endValue) {
            IRange result = new ArcScaleRange();
            result.AppearanceRange.ContentBrush = new SolidBrushObject(contentColor);
            result.StartValue = startValue;
            result.EndValue = endValue;
            result.StartThickness = 0f;
            result.EndThickness = 50f;
            return result;
        }
    }
}

File: VB/Q205964/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGauges.Win
Imports DevExpress.XtraGauges.Win.Gauges.Circular
Imports DevExpress.XtraGauges.Base
Imports DevExpress.XtraGauges.Core.Model
Imports DevExpress.XtraGauges.Core.Drawing

Namespace Q205964
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			CreateCircularGauge()
		End Sub

		Private Sub CreateCircularGauge()
			Dim gaugeControl As New GaugeControl()
			Dim gauge As CircularGauge = CType(gaugeControl.AddGauge(GaugeType.Circular), CircularGauge)
			gauge.AddDefaultElements()
			Dim scale As ArcScaleComponent = gauge.Scales(0)
			scale.BeginUpdate()
			scale.Ranges.AddRange(CreateRanges(scale.MaxValue - scale.MinValue, scale.MinValue))
			scale.EndUpdate()
			gaugeControl.Dock = DockStyle.Fill
			Controls.Add(gaugeControl)
		End Sub

		Private rangeContentColors() As Color = { Color.Red, Color.Yellow, Color.Green }

		Private Function CreateRanges(ByVal ticks As Single, ByVal start As Single) As IRange()
			Dim range As Integer = CInt(Fix(Math.Floor(CDbl(ticks / rangeContentColors.Length))))
			Dim ranges As New List(Of IRange)()
			Dim i As Integer = 0
			Do While i < rangeContentColors.Length
'INSTANT VB TODO TASK: Assignments within expressions are not supported in VB.NET
'ORIGINAL LINE: ranges.Add(CreateRange(rangeContentColors[i], (int)(range * i + start), (int)(range * ++i + start)));
				ranges.Add(CreateRange(rangeContentColors(i), CInt(Fix(range * i + start)), CInt(Fix(range * ++i + start))))
			Loop
			Return ranges.ToArray()
		End Function

		Private Function CreateRange(ByVal contentColor As Color, ByVal startValue As Integer, ByVal endValue As Integer) As IRange
			Dim result As IRange = New ArcScaleRange()
			result.AppearanceRange.ContentBrush = New SolidBrushObject(contentColor)
			result.StartValue = startValue
			result.EndValue = endValue
			result.StartThickness = 0f
			result.EndThickness = 50f
			Return result
		End Function
	End Class
End Namespace